src/SAL/nfs4_recovery.c

```diff
@@ -1050,23 +1051,10 @@ static void nfs_release_nlm_state(char *release_ip)
 
 static int ip_match(char *ip, nfs_client_id_t *cid)
 {
-	char *haystack;
-	char *value = cid->cid_client_record->cr_client_val;
-	int len = cid->cid_client_record->cr_client_val_len;
+	in_addr_t saddr = inet_addr(ip);
+	saddr = ntohl(saddr);
 
-	LogDebug(COMPONENT_STATE, "NFS Server V4 match ip %s with (%.*s)",
-		 ip, len, value);
-
-	if (strlen(ip) == 0)	/* No IP all are matching */
-		return 1;
-
-	haystack = alloca(len + 1);
-	memcpy(haystack, value, len);
-	haystack[len] = '\0';
-	if (strstr(haystack, ip) != NULL)
-		return 1;
-
-	return 0;		/* no match */
+	return (uint32_t)saddr == cid->server_addr;
```

Comment:
server_addr is obtained by invoking `uint64_t client_record_value_hash(nfs_client_record_t *key)`. Does it have the same logic as `inet_addr`?